sunpy/net/dataretriever/sources/goes.py

```diff
@@ -223,183 +85,86 @@ class SUVIClient(GenericClient):
-                elif this_wave == 94:
-                    search_pattern = base_url + \
-                        r'l{level}/suvi-l{level}-fe{wave:03}/%Y/%m/%d/OR_SUVI-L{level}-Fe{wave_minus1:03}_G{goes_number}_s%Y%j%H%M%S.*\.fits.gz'
-
-            if search_pattern.count('wave_minus1'):
-                scraper = Scraper(search_pattern, level=level, wave=this_wave,
-                                  goes_number=satellitenumber, wave_minus1=this_wave-1)
-            else:
-                scraper = Scraper(search_pattern, level=level, wave=this_wave,
-                                  goes_number=satellitenumber)
-            results.extend(scraper.filelist(timerange))
-        return results
-
-    def _makeimap(self):
-        """
-        Helper Function used to hold information about source.
-        """
-        self.map_['source'] = 'GOES'
-        self.map_['provider'] = 'NOAA'
-        self.map_['instrument'] = 'SUVI'
-        self.map_['physobs'] = 'flux'
-
-    @classmethod
-    def _can_handle_query(cls, *query):
-        """
-        Answers whether client can service the query.
-
-        Parameters
-        ----------
-        query : `tuple`
-            All specified query objects.
-
-        Returns
-        -------
-        `bool`
-            answer as to whether client can service the query.
-        """
-        # Import here to prevent circular imports
-        from sunpy.net import attrs as a
-
-        required = {a.Time, a.Instrument}
-        optional = {a.Wavelength, a.Level, a.goes.SatelliteNumber}
-        all_attrs = {type(x) for x in query}
-
-        ops = all_attrs - required
-        # check to ensure that all optional requirements are in approved list
-        if ops and not all(elem in optional for elem in ops):
-            return False
-
-        # if we get this far we have either Instrument and Time
-        # or Instrument, Time and Wavelength
-        check_var_count = 0
-        for x in query:
-            if isinstance(x, a.Instrument) and x.value.lower() == 'suvi':
-                check_var_count += 1
-
-        if check_var_count == 1:
-            return True
+    baseurl1b = (r'https://data.ngdc.noaa.gov/platforms/solar-space-observing-satellites/goes/goes'
+                 r'{SatelliteNumber}/l1b/suvi-l1b-{elem:2}{wave:03}/%Y/%m/%d/OR_SUVI-L1b.*\.fits.gz')
+    pattern1b = ('{}/goes/goes{SatelliteNumber:2d}/l{Level:2w}/suvi-l1b-{}{Wavelength:03d}/'
+                 '{year:4d}/{month:2d}/{day:2d}/{}_s{:7d}{hour:2d}{minute:2d}{second:2d}'
+                 '{:1d}_e{:7d}{ehour:2d}{eminute:2d}{esecond:2d}{:1d}_{}')
+    baseurl2 = (r'https://data.ngdc.noaa.gov/platforms/solar-space-observing-satellites/goes/goes{SatelliteNumber}/'
+                r'l2/data/suvi-l2-ci{wave:03}/%Y/%m/%d/dr_suvi-l2-ci{wave:03}_g{SatelliteNumber}_s%Y%m%dT%H%M%SZ_.*\.fits')
+    pattern2 = ('{}/goes/goes{SatelliteNumber:2d}/{}/dr_suvi-l{Level}-ci{Wavelength:03d}_g{SatelliteNumber:2d}_s'
```

Comment:
In theory this could be reduced if the scrapper was smarter?

Reply:
Yeah, only convertors need to befined. (to convert level 2 to `/data/l2/` in baseurl). We can do everything with one baseurl itself.